build-system/tasks/runtime-test/helpers-test-reporting.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const fetch = require('node-fetch');
+const fs = require('fs');
+const log = require('fancy-log');
+const JSON_REPORT_TEST_TYPES = new Set(['unit', 'integration']);
+const REPORTING_API_URL = 'https://amp-test-cases.uc.r.appspot.com/report';
+const {
+  travisBuildNumber,
+  travisJobNumber,
+  travisCommitSha,
+} = require('../../common/travis');
+
+function resultFilename(testType) {
+  return `results_${testType}.json`;
+}
+
+async function getReports() {
+  const reportPromises = JSON_REPORT_TEST_TYPES.map(async (testType) => {
+    const jsonString = await fs.readFile(resultFilename(testType));
+    const reportJson = JSON.parse(jsonString);
+    return {testType, reportJson};
+  });
+
+  const incompleteReports = await Promise.all(reportPromises);
+  const reports = incompleteReports.map(addJobAndBuildInfo);
+
+  return reports;
+}
+
+function addJobAndBuildInfo({testType, reportJson}) {
+  const build = {
+    buildNumber: travisBuildNumber(),
+    commitHash: travisCommitSha(),
+    startedAt: 0,
+    jobs: [],
+  };
+
+  const job = {
+    jobNumber: travisJobNumber(),
+    testSuiteType: testType,
+    startedAt: 0,
+    testRuns: [],
```

Comment:
Seems like this is unnecessary if the actual results are in the `results` property?

Reply:
The `testRuns` array is there so that the JSON fits the Job interface at https://github.com/ampproject/amp-github-apps/blob/master/test-case-reporting/types/test-case-reporting.d.ts. Same above with the `jobs` array in the build.

For the job, we could get an approximate `startedAt` from the Karma test report data, by taking the start time of the first test in the report. I don't believe we could do the same for the build.

Reply:
We could just make a new type for the Build & Job info given by Travis, that way we don't need to send the empty list and the invalid startedAt (which could be misleading).

Reply:
Done.